packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -74,6 +90,56 @@ function readTreeSection(node: NodeCore) {
     };
     for (const treeNode of node) {
         assertNodeCoreInstance(treeNode, "tree nodes should be nodes");
+
+        /**
+         * Perf optimization - the most common cases!
+         * This is essentially unrolling code below for faster processing
+         * It speeds up tree parsing by 2-3x times!
+         */
+        if (treeNode.getMaybeString(0) === "name") {
```

Comment:
Are we sure that we will always have 1 element as we are accessing the 0th element directly?

Reply:
We should check the length before getting this element. Rest looks fine to me.

Reply:
Good point! I'll reorder code not to have that issue :)

Reply:
That said, it works Ok as we get undefined for such cases :)